src/aten/Distributions.cpp

```diff
@@ -0,0 +1,221 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/Dispatch_v2.h>
+#include <ATen/ScalarOps.h>
+#include <ATen/XPUNativeFunctions.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/native/DistributionTemplates.h>
+#include <ATen/native/TensorIterator.h>
+#include <aten/sycl/DistributionKernels.h>
+
+namespace at {
+
+Tensor& XPUNativeFunctions::normal_(
+    Tensor& self,
+    double mean,
+    double std,
+    ::std::optional<at::Generator> generator) {
+  TORCH_CHECK(std >= 0.0, "normal_ expects std >= 0.0, but found std=", std);
+  if (self.is_complex()) {
+    auto float_tensor = at::view_as_real(self);
+    // variance for normal distribution of the real and imaginary values
+    // is half of the input variance
+    native::xpu::normal_kernel(
+        float_tensor, mean, std / (std::sqrt(2)), generator);
+  } else {
+    native::xpu::normal_kernel(self, mean, std, generator);
+  }
+  return self;
+}
+
+#define CHECK_OUT_OF_BOUNDS(var, name, min, max, dtype) \
+  TORCH_CHECK(var >= min && var <= max, name, " is out of bounds for ", dtype);
+
+#define CHECK_EMPTY_AND_RETURN(tensor) \
+  if (tensor.numel() == 0) {           \
+    return tensor;                     \
+  }
+
+Tensor& XPUNativeFunctions::uniform_(
+    Tensor& self,
+    double from,
+    double to,
+    ::std::optional<at::Generator> generator) {
+  if (self.is_complex()) {
```

Comment:
Can we reuse the template ?
<img width="944" alt="image" src="https://github.com/intel/torch-xpu-ops/assets/28250760/820dd625-a3d4-46d6-802a-624f61310128">